### PR TITLE
Add the option of use only SSO login and register

### PR DESCRIPTION
**Pull request by melvinsoft**
The PR introduces the ability of redirect the login and register page AKA logistration to an SSO Identity Provider, this is more efficient than using Nginx rules as we currently do.

This is a temporary patch, since we already have a [ProdPad idea](https://app.prodpad.com/ideas/93b22e30-e046-11e7-9e86-8f01e6a8263f/canvas) to implements this upstream first and then start using it.

How to test it:
1. Checkout this branch.
2. Add a new feature flag under `APPSEMBLER_FEATURES`:
```
"ENABLE_EXCLUSIVE_SSO_LOGISTRATION": true
```
3. Add a new Env variable with a url map:
```
"EXCLUSIVE_SSO_LOGISTRATION_URL_MAP": {
        "register": "http://appsembler.com",
        "login": "http://appsembler.com"
    },
```

### CODE REVIEW

**Comment by melvinsoft** [comment]
[(3) Trinity: Investigates new users without SSO link.](https://trello.com/c/KMOmzBwC/2280-3-trinity-investigates-new-users-without-sso-link)

File: lms/envs/aws.py [hunk 1]
```diff
@@ -870,3 +870,5 @@
 APP_UPGRADE_CACHE_TIMEOUT = ENV_TOKENS.get('APP_UPGRADE_CACHE_TIMEOUT', APP_UPGRADE_CACHE_TIMEOUT)
 
 AFFILIATE_COOKIE_NAME = ENV_TOKENS.get('AFFILIATE_COOKIE_NAME', AFFILIATE_COOKIE_NAME)
+
+EXCLUSIVE_SSO_LOGISTRATION_URL_MAP = ENV_TOKENS.get('EXCLUSIVE_SSO_LOGISTRATION_URL_MAP', {})
```
Comment (OmarIthawi) [inline lms/envs/aws.py]: @melvinsoft you've used `aws.py`, but `devstack_appsembler.py`. Care to unify?

Comment (melvinsoft) [reply lms/envs/aws.py]: @OmarIthawi Thanks, change addressed.

**Comment by OmarIthawi** [comment]
Thanks @melvinsoft! Looking forward to get this merged!

**Pull request merged**